examples/starknet-react-next/src/components/DepositToken.tsx

```diff
@@ -0,0 +1,76 @@
+import { useStarknet, useStarknetExecute } from '@starknet-react/core'
+import { useState, useCallback, useMemo } from 'react'
+import { toBN } from 'starknet/dist/utils/number'
+
+export function DepositToken() {
+  const { account } = useStarknet()
+  const [amount, setAmount] = useState('')
+  const [amountError, setAmountError] = useState<string | undefined>()
+
+  const tokenAddress = '0x07394cbe418daa16e42b87ba67372d4ab4a5df0b05c6e554d158458ce245bc10'
+  const bankAddress = '0x0185a8709eedfd3da1a0d4a0739e649ddbe9433a4ca32f63de8824e52f078cda'
+
+  const updateAmount = useCallback(
+    (newAmount: string) => {
+      // soft-validate amount
+      setAmount(newAmount)
+      try {
+        toBN(newAmount)
+        setAmountError(undefined)
+      } catch (err) {
+        console.error(err)
+        setAmountError('Please input a valid number')
+      }
+    },
+    [setAmount]
+  )
+
+  const calls = [
+    {
+      contractAddress: tokenAddress,
+      entrypoint: 'approve',
+      calldata: [toBN(bankAddress).toString(), amount, '0'],
```

Comment:
I assume this is because the amount is an uint256? Should work with `toBN(amount)` instead of `amount, '0'`.